Upload blog post edit (#19)
CorneliaKelinske/first_blog: posts/2021/11-28-media_upload_to_database.md

Before we get down to the nitty-gritty, let's make sure we are on the same page. I am assuming that you have set up your Phoenix project including your repo and your uploads and user schemas.  

> For reference, my project has an accounts context in which I am defining my user schema and my upload schema is defined in my content context. Accordingly, my '/lib/my_project folder' has the following structure:

Reviewer: ```suggestion
For reference, my project has an accounts context for my user schema and a content context for my upload schema. Accordingly, my '/lib/my_project folder' has the following structure:
```

│   ├── my_project.ex

```
And then I have the usual suspects: controllers, views, templates in the '/lib/my_project_web' folder.

# 4. Upload schema

> The first step was to define my 'uploads' schema in my 'upload.ex' file. What's notable in this regard is that I am storing the  image/video file as binary data.

Reviewer: ```suggestion
The first step was to define my `uploads` schema in my 'upload.ex' file. What's notable in this regard is that I am storing the image/video file as binary data.
```
Use backticks (above the tilde) for inline code

I also added module attributes specifying valid image types and valid video types as well as a combination of both. In order to be able to use the valid image and video types in my templates, I created corresponding functions returning the valid file types.

> I am using my 'changeset/2' function to verify that the file type of the file the user wants to upload is included in the valid file types.

Reviewer: ```suggestion
I am using my `changeset/2` function to verify that the file type of the file the user wants to upload is included in the valid file types.
```

  end
end
```

> And just to be extra thorough: my 'content.ex' file includes a 'create_upload/1' function, which I will not elaborate on, since it is one of the functions that the Phoenix generator generates by default.

Reviewer: ```suggestion
And just to be extra thorough: my `content.ex` file includes a `create_upload/1` function, which I will not elaborate on, since it is one of the functions that the Phoenix generator generates by default.
```

```

> Let's look at the upload. Our initial upload in the 'create/2' function is what the user inputs into the 'new upload' form under '/uploads/new'and what is subsequently sent through the form to the upload controller. It looks like this:

Reviewer: I won't fix any more backticks, I think you get the idea :)

> Let's look at the upload. Our initial upload in the 'create/2' function is what the user inputs into the 'new upload' form under '/uploads/new'and what is subsequently sent through the form to the upload controller. It looks like this:

Reviewer: ```suggestion
Let's look at the upload. Our initial upload in the 'create/2' function is what the user inputs into the 'new upload' form under '/uploads/new' and what is subsequently sent through the form to the upload controller. It looks like this:
```

```

# 8. But will it scale?!

> No. This is a solution for small, personal projects. It is not meant to scale. It's a neat approach when you are just starting to work on something and don't want to enter into any commitment with third-party providers (yet) or when you know that your project will indeed remain a small one.

Reviewer: A bit harsh to just say "No.".

Maybe: "It sure won't!"
